### ISSUE TITLE
train_classification.py中如何应对测试集中超过100个样本的情况？

### ISSUE THREAD

**Issue by HU13-SVG** [sjtuplayer/anomalydiffusion#107]
def test(args,obj_name, model,anomaly_names):
    model.eval()

    dataset = MVTec_classification_test(args,obj_name,anomaly_names)
    dataloader = DataLoader(dataset, batch_size=100,
                            shuffle=False, num_workers=0)

    for i_batch, sample_batched in enumerate(dataloader):
        image, label = sample_batched
        image = image.cuda()
        label = label.cuda()
        y_pred = model(image)
        prediction = torch.argmax(y_pred, 1)
        correct = (prediction == label).sum().float()
        print("Accuracy: %.4f"%(correct/len(label)))
    return correct/len(label)
# 返回的是最后一个batch的准确率

**Comment by sjtuplayer** [comment]
> def test(args,obj_name, model,anomaly_names): model.eval()
> 
> ```
> dataset = MVTec_classification_test(args,obj_name,anomaly_names)
> dataloader = DataLoader(dataset, batch_size=100,
>                         shuffle=False, num_workers=0)
> 
> for i_batch, sample_batched in enumerate(dataloader):
>     image, label = sample_batched
>     image = image.cuda()
>     label = label.cuda()
>     y_pred = model(image)
>     prediction = torch.argmax(y_pred, 1)
>     correct = (prediction == label).sum().float()
>     print("Accuracy: %.4f"%(correct/len(label)))
> return correct/len(label)
> ```
> 
> # 返回的是最后一个batch的准确率

You can just change the batch_size to the size you need.
